### PR TITLE
Feature/donation form ect

### PR DESCRIPTION
**Pull request by ectweitmann**
### 1. What changed?
- Created the DonationFormModal component.

### 2. Why is this change necessary?
- Our application needed a form through which a pet owner could submit pet's information and put them up for adoption.

### 3. What changed technically that may impact others?
- This is a self-contained PR at this time and does not impact other components/parts of the code base.
- However we will need to go back and add functionality to the file input field as well as add the functionality to close the modal.

### 4. How do we test it?
- Considering that the Home page view is not currently set up, to test/see the DonationFormComponent at this time, you need to go into `App.js` and create a temporary route which displays the `DonationFormComponent`. After that run `npm start` navigate to the page using the route path you created, and open the dev tools. After that fill in the form and make sure that formData state is updated upon inputing data.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -3,6 +3,10 @@
   <head>
     <meta charset="utf-8" />
     <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <link rel="preconnect" href="https://fonts.googleapis.com">
```
Comment (anewb87) [inline public/index.html:6]: Always nice when you can find the exact font used on your wireframe.

File: src/Components/DonationFormModal.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
+import React, { useState } from 'react';
+import '../Styles/DonationFormModal.scss';
+
+const DonationFormModal = () => {
+  const [formState, setFormState] = useState({
```
Comment (anewb87) [inline src/Components/DonationFormModal.js:5]: Our first hook.

File: src/Styles/DonationFormModal.scss [hunk 3]
```diff
@@ -0,0 +1,87 @@
+@import './_variables';
+
```
Comment (anewb87) [inline src/Styles/DonationFormModal.scss:2]: Love the look of this modal/form.

**Review by anewb87 — APPROVED** [review]
Love how this modal/form is coming along. Forms can be a bit...tedious but feel like we worked together well to deepen our understanding of them, add a hook, learn new things (like textarea and some css properties I'd never used quite in the same way). Excited to keep at it and figure out the dropdown bug we haven't yet squared away here in React.

**Review by omegaeye-he — COMMENTED** [review]
Looks like this form takes a lot of work to put together. Is there a way to divide this PR in smaller chunks? and each chunk is complete and tested.

**Comment by ectweitmann** [comment]
@omegaeye-he I'm not sure I understand. Could you elaborate on what you are suggesting?

File: src/Components/DonationFormModal.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+          <h3 className="form-subtitle">Owner Information</h3>
+          <div className="owner-name-container">
+            <div className="input-label">
+              <label htmlFor="firstName">First Name</label>
+              <input
+                type="text"
+                value={formState.firstName}
+                id="firstName"
+                placeholder="e.g. Samantha"
+                onChange={(e) =>
```
Comment (StevieAmb) [inline src/Components/DonationFormModal.js:30]: Great job using hooks to set the state immediately as the user is typing, using hooks.

File: src/Components/DonationFormModal.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+                  imageUpload: e.target.value
+                })
+              }
+              required
+            />
+          </div>
+        </section>
+        <button className="submit-button">Submit</button>
+      </form>
+    </div>
```
Comment (StevieAmb) [inline src/Components/DonationFormModal.js:180]: Amazing job with creating this modal and working out the functionality of it.

**Review by StevieAmb — APPROVED** [review]

File: src/Components/DonationFormModal.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
+import React, { useState } from 'react';
+import '../Styles/DonationFormModal.scss';
+
+const DonationFormModal = () => {
+  const [formState, setFormState] = useState({
+    firstName: '',
```
Comment (Fordo29) [inline src/Components/DonationFormModal.js:6]: Question... I know I was the one to suggested the route to doing a hook like this, but other sources have them as their own const.. Such as const [firstName, setFirstName] = useState('').  I am not sure which looks cleaner

**Review by Fordo29 — APPROVED** [review]

**Pull request merged**